### PR TITLE
Add go-based TokenDuality e2e test

### PR DESCRIPTION
**Pull request by karlb**
Unfortunately the test still has two problems. Please provide feedback on these issues!

## Uses external `geth` command

The geth command is not automatically built when running `go test`, so it might be outdated or completely missing.

Potential solutions:
* Run `make geth` from inside the test. I feel like this is something a test should not do and will surprise users of the test suite who might rely on the last explicitly built geth version for something else they do.
* Use the test binary. This is something which is [done in `./cmd/geth`](https://github.com/celo-org/op-geth/blob/e44290ac3a9eded9d3b5ae6476779d5f15c86217/cmd/geth/run_test.go#L60-L86). But without major changes, it will only work if we move the e2e test into that package, as not only private functions are used to run the test binary, but also specific code has to go into the test binary to execute geth from there.

## `geth` not reliably killed after test

Using the `CommandContext` should send a SIGKILL to geth when the test ends (due to the deferred call to the call's cancel function), but sometimes geth survives and keeps operating like it never received such a signal. I witnessed occasional left over geth processed from other parts of the OP-stack, so maybe the test code is find, but geth is hard to kill in some unusual way?

Potential solutions:
* Thoroughly check why this is happening and if geth reliably reacts to SIGKILL outsite of the test.
* At the end of the context (or in another deffered), check if the process still exists and kill it repeatedly until it is gone.

### CODE REVIEW

File: core/celo_genesis.go [hunk 1]
```diff
@@ -36,25 +40,46 @@ func celoGenesisAccounts() map[common.Address]GenesisAccount {
...
+				proxy_implementation_slot: common.HexToHash("0xce13"),
+				proxy_owner_slot:          DevAddr,
+			},
+			Balance: big.NewInt(0),
+		},
+		common.HexToAddress("0xce13"): { // GoldToken Implementation
+			Code:    goldTokenBytecode,
+			Balance: big.NewInt(0),
+		},
+		common.HexToAddress("0x42cf1bbc38BaAA3c4898ce8790e21eD2738c6A4a"): {
```
Comment (carterqw2) [inline core/celo_genesis.go]: Why not reuse `DevAddr`?

**Review by carterqw2 — COMMENTED** [review]
What if we externalize geth management completely and accept the endpoint as an environment variable instead? We can provide a command/script to do that locally, and in CI we might want to implement that differently anyway.

File: core/celo_genesis.go (hunk 1 above)
Comment (karlb) [reply core/celo_genesis.go]: I just missed that occurrence of the address when introducing DevAddr. I changed i just now, but I had to distinguish between the real DevAddr and the 32byte representation of it for the storage fields.

**Comment by karlb** [comment]
> What if we externalize geth management completely and accept the endpoint as an environment variable instead? We can provide a command/script to do that locally, and in CI we might want to implement that differently anyway.

That would prevent us from getting a fresh state for each test. But on the bright side, using the same geth across multiple  tests makes the tests run faster. We could write the test in a way that it works on unclean state and provide a helper (e.g. a make target) to run the tests as expected with a freshly compiled running geth instance.

How/when to stop geth might still be somewhat tricky.

**Pull request closed without merging**